tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
```

Comment:
"predictable enough" means it really only has the possibility to vary if there is more than just the test suite hitting the server. It would probably be smart for us to find a way to isolate these tests from the firebase server somehow. That would remove a lot of moving parts.